File: library/src/main/java/com/getstream/sdk/chat/enums/QuerySortAdapter.java
package com.getstream.sdk.chat.enums;

import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;
import java.util.ArrayList;

public class QuerySortAdapter extends TypeAdapter<QuerySort> {
    @Override
    public void write(JsonWriter out, QuerySort value) throws IOException {
        TypeAdapter adapter = new Gson().getAdapter(ArrayList.class);
        adapter.write(out, value.getData());
    }

    @Override
    public QuerySort read(JsonReader in) throws IOException {
        throw new IOException("not supported");
    }
}

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser/adapters/QuerySortAdapter.kt
package io.getstream.chat.android.client.parser.adapters

import com.google.gson.Gson
import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import io.getstream.chat.android.client.api.models.QuerySort
import java.io.IOException
import java.util.ArrayList

internal class QuerySortAdapter(
    private val gson: Gson
) : TypeAdapter<QuerySort<*>>() {
    override fun write(out: JsonWriter, value: QuerySort<*>?) {
        val adapter = gson.getAdapter(ArrayList::class.java)
        adapter.write(out, value?.toMap() as? ArrayList<*>)
    }

    override fun read(`in`: JsonReader?): QuerySort<*>? {
        throw IOException("QuerySort must not be deserialized")
    }
}
